File: 2022-23/contest1/prog/junior/next_base.py
import os
import sys

def str_to_int(s, base):
	res = 0
	N = len(s)
	for i in range(N):
		res += ((ord(s[N-i-1]) - ord('0')) * pow(base, i))
	return res

def digit_sum(num, base):
	res = 0
	while num > 0:
		d = num % base
		res += d
		num = num // base
	return res

for k in range(5):
	args = [x for x in input().split()]
	n = int(args[0])
	b = int(args[1])
	s = args[2]
	num = str_to_int(s, b)
	tot = 0
	for i in range(n):
		ds = digit_sum(num+i, b)
		tot += ds;
	print(tot)

File: 2022-23/contest1/prog/intermediate/next_base.cpp
#include <bits/stdc++.h>
using namespace std;

long long str_to_int(string str, long long base) {
	long long res = 0;
	for(long long i = 0, N = str.length(); i < N; ++i) {
		res += ((str[N-i-1] - '0') * (long long)(pow(base, i)));
	}
	return res;
}

int main() {
	for(int k = 0; k < 5; ++k) {
		long long n,b;
		string s;
		cin >> n >> b >> s;
		auto num = str_to_int(s, b);
		string nums;
		int cnts[10] = {};
		for(int i = 0; i < n; ++i) {
			long long numi = num + i;
			while(numi) {
				long long d = numi % b;
				++cnts[d];
				numi /= b;
			}
		}
		cout << cnts[b-1] << endl;
	}
	return 0;
}
